### PR TITLE
Plug a packet buffer memory leak

### PR DESCRIPTION
**Pull request by stevenmhood**
Decrement reference count after queueing the PacketBuffer.  Since PacketBuffer objects come from a pool, this allows them to be returned to the pool instead of staying in memory and forcing new allocations.

Fixes https://github.com/containers/gvisor-tap-vsock/issues/107

My interest in fixing this lies in this projects downstream usage in wsl-vpnkit, which grows its memory footprint until my system comes to a standstill (several GB of RAM).  With this fix, memory usage stays around 20MB, give or take.  Tracing the `NewPacketBuffer` call back through the stack exposed the `Pool` that underlies the `PacketBuffer` instances.

### CODE REVIEW

**Review by cfergeau — APPROVED** [review]
This matches the way it's used in the gvisor codebase, looks good to me. Thanks a lot for the patch!

**Comment by cfergeau** [comment]
This is missing a DCO before this can be merged, https://github.com/containers/gvisor-tap-vsock/pull/152/checks?check_run_id=9081855178

**Comment by stevenmhood** [comment]
Moved this to re-request, now that the DCO passed, since the other workflows need approval to run.  Not sure if you check these periodically or on-demand, either way, no rush from my side.  Thanks!

**Pull request merged**
